### PR TITLE
Feature/bit 103

### PR DESCRIPTION
**Pull request by Chrissydrx**
This pull request contains all the backend logic to send mails. There are four different mails:
- send Login Links to all users
- send a Mail before the BIT to remind the user for the date and also his decision
- send a reminder Mail to the user to decide for two professional fields and the general presentation (Just to users who didn't make a decision)
- send to an user a new Login Link if the first_name and surname are entered correctly

This branch also includes an updated composer file. Mailhog now will be installed and block all mails to be sent (just test purpose). You can find an advice to use it properly in the readme file.

### CODE REVIEW

**Comment by julix14** [comment]
Can you @Chrissydrx please fix the DeepSource errors?

File: app/Http/Controllers/EmailController.php [hunk 1]
```diff
@@ -0,0 +1,80 @@
+<?php
+
+namespace App\Http\Controllers;
+
+use App\Mail\decisionReminderMail;
```
Comment (julix14) [inline app/Http/Controllers/EmailController.php]:
```suggestion
use App\Mail\DecisionReminderMail;
```

File: README.md [hunk 2]
```diff
@@ -23,3 +23,15 @@ Für die tägliche Entwicklungsarbeit müssen jeweils folgende Schritte durchgef
 * `sail artisan migrate:fresh` setzt die Datenbank zurück und baut eine neue auf
 * `sail artisan migrate` setzt eine neue Datenbank auf
 * `sail artisan db:seed` speichert die Testwerte in die Datenbank
+
+## Mailhog benutzen
+* Folgendes muss in die .env datei
```
Comment (julix14) [inline README.md]:
```suggestion
* Folgendes muss in die .env Datei 
```

File: app/Http/Controllers/EmailController.php [hunk 3]
```diff
@@ -0,0 +1,80 @@
...
+
+    //Send a mail before the next BIT to remind the user for the date and also about his decision
+    public function sendBeforeBITMailToAllUsers(){
+        $users = User::all();
+        foreach($users as $user) {
+            $this->sendBeforeBITMail($user);
+        }
+    }
+
+    //Send a mail to all users, who doesn't make a decision
```
Comment (julix14) [inline app/Http/Controllers/EmailController.php:32]:
```suggestion
    //Send a mail to all users, who doesn't made a decision
```

File: app/Http/Controllers/EmailController.php [hunk 4]
```diff
@@ -0,0 +1,80 @@
...
+
+    private function sendDecisionReminderMail(User $user){
+        Mail::to($user->email)->send(new DecisionReminderMail($user));
+    }
+
+    //Helper functions
+    private function userMadeDecision(User $user):bool{
+        //If the system can find an entry in professionalFieldDecision, the system knows that the user made a complete decision, because it is not possible to just select one professionalField
+        $professionalFieldDecision1 = ProfessionalFieldDecision::where('user_id', $user->id)->first();
+        if(isset($professionalFieldDecision1)){
```
Comment (julix14) [inline app/Http/Controllers/EmailController.php]:
```suggestion
        $anyProfessionalFieldDecision = ProfessionalFieldDecision::where('user_id', $user->id)->first();
        if(isset($anyProfessionalFieldDecision){
```

Why "1" if there is only one? I would assume there is a second one

File: app/Mail/LoginLinkMail.php [hunk 5]
```diff
@@ -0,0 +1,67 @@
...
+
+    /**
+     * Get the message envelope.
+     *
+     * @return \Illuminate\Mail\Mailables\Envelope
+     */
+    public function envelope()
+    {
+        return new Envelope(
+            subject: 'Dein persönlicher Login-Link',
```
Comment (julix14) [inline app/Mail/LoginLinkMail.php:37]: It would be good to name the "BIT" i think. so the recipient knows what login-link he received

Comment (Chrissydrx) [reply app/Mail/LoginLinkMail.php:37]: The design of the mail is still work in progress, because it is your work to create a proper email :)

Comment (julix14) [reply app/Mail/LoginLinkMail.php:37]: _Votekick_ @Chrissydrx

**Review by julix14 — APPROVED** [review]
Super gemacht 😘😘

**Pull request merged**